### PR TITLE
fix softmax precision error axis!=-1.

### PR DESCRIPTION
**Pull request by heavengate**
**fix precision bugs in softmax when axis!=-1**
shifted_logits calculate max values only along axis.

### CODE REVIEW

**Review by JesseyXujin — APPROVED** [review]

**Review by kuke — APPROVED** [review]
LGTM
